src/document.rs

```diff
@@ -552,6 +554,24 @@ impl Document {
     fn __repr__(&self) -> PyResult<String> {
         Ok(format!("{self:?}"))
     }
+
+    fn __copy__(&self) -> Self {
+        self.clone()
+    }
+
+    fn __deepcopy__(&self, _memo: &PyDict) -> Self {
+        self.clone()
+    }
+
+    fn __richcmp__(&self, other: &Self, op: CompareOp) -> PyResult<bool> {
+        match op {
+            CompareOp::Eq => Ok(self == other),
+            CompareOp::Ne => Ok(self != other),
+            _ => Err(PyNotImplementedError::new_err(format!(
+                "{op:?} op is not supported"
+            ))),
+        }
```

Comment:
I believe the more expected thing is to return `NotImplemented` for unimplemented Ops.  This is discussed briefly in the PyO3 docs:

https://pyo3.rs/v0.18.3/class/object.html?highlight=alternatively#comparisons

![image](https://github.com/quickwit-oss/tantivy-py/assets/480395/c88ec735-c3ca-4198-ba59-cfbbe83b135e)